File: longest_word/longest_word.py
"""
>>> findLongestWord("The quick brown fox jumped over the lazy dog")
6
>>> findLongestWord("May the force be with you")
5
>>> findLongestWord("Google do a barrel roll")
6
>>> findLongestWord("What is the average airspeed velocity of an unladen swallow")
8
>>> findLongestWord("What if we try a super-long word such as otorhinolaryngology")
19
"""

def findLongestWord(string: str) -> int:
    """

    :param string:
    :return: integer
    """

    # initialize longest word variable
    longest_word = 1

    # split string in a list of strings
    words_array = string.split(" ")

    # compare each word with the value of longest_word
    for word in words_array:
        # assign new value to longest_word if word length is greater than longest_word
        if len(word) > longest_word:
            longest_word = len(word)

    return longest_word

File: longest_word/longest_word.js
// Uses JS 5 syntax

function findLongestWord(str) {
  "use strict";
  // initalize longest word variable
  var longestWord = 1;

  // split str by space break
  var splitter = str.split(' ');
  // loop through array of words
  splitter.forEach(function(word){
    // compare current word length with 'longestWord' length
    if (word.length > longestWord) {
      longestWord = word.length;
    }
  });
  console.log(longestWord);
  return longestWord;
}

findLongestWord("The quick brown fox jumped over the lazy dog");
findLongestWord("May the force be with you");
findLongestWord("Google do a barrel roll");
findLongestWord("What is the average airspeed velocity of an unladen swallow");
findLongestWord("What if we try a super-long word such as otorhinolaryngology");


// Uses JS 6 syntax

const findLongestWordES6 = str => {
  // initalize longest word variable
  let longestWord = 1;

  // split str by space break
  const splitter = str.split(' ');
  // loop through array of words
  splitter.forEach(function(word){
    // compare current word length with 'longestWord' length
    if (word.length > longestWord) {
      longestWord = word.length;
    }
  });
  console.log(`This is ES6 ---> ${longestWord}`);
  return longestWord;
};

findLongestWordES6("The quick brown fox jumped over the lazy dog");
findLongestWordES6("May the force be with you");
findLongestWordES6("Google do a barrel roll");
findLongestWordES6("What is the average airspeed velocity of an unladen swallow");
findLongestWordES6("What if we try a super-long word such as otorhinolaryngology");